gslib/utils/user_agent_helper.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains helper for appending user agent information."""
+
+import sys
+import gslib
+from gslib.utils import system_util
+from gslib.utils.arg_helper import GetArgumentsAndOptions
+
+
+def GetUserAgent(metrics_off=True):
+  """Using the command arguments return a suffix for the UserAgent string.
+
+  Args:
+    metrics_off: boolean, whether the MetricsCollector is disabled.
+
+  Returns:
+    str, A string value that can be appended to an existing UserAgent.
+  """
+  _, opts = GetArgumentsAndOptions()
+
+  user_agent = ' gsutil/%s' % gslib.VERSION
+  user_agent += ' (%s)' % sys.platform
+  user_agent += ' analytics/%s ' % ('disabled' if metrics_off else 'enabled')
+  user_agent += ' interactive/%s' % sys.stdin.isatty()
+
+  if len(opts) > 0:
+    user_agent += ' command/%s' % opts[0]
+    if len([segment for segment in opts if '://' in segment]) > 1:
+      user_agent += '-CloudToCloud'
```

Comment:
For a cp command, the source list can be a list of local file paths as well as cloud paths. In that case, won't we end up setting this as `CloudToCloud` for local file upload/download as well?
Also, it doesn't cover cases where a file path follows the uri `file:///...` which might be a file on a network but not necessarily on gcs or s3.

Reply:
This is a "cheap" way of figuring out if the operations are cloud to cloud with some significant limitations. Actually calculating the real destinations here seemed like it would be out of reach, given the complexity of that operation and the fact that it would need to happen right after bootstrap. I hadn't honestly considered the prospect of `file:///` prefixes, so I'll narrow the scope to only gs:// and s3:// prefixes.

Reply:
Yeah, I can see that figuring out the CloudToCloud flag for each src destn pair might become complicated and given that the number of cases where users mix local path with cloud path in the source url list for the cp command might be low, this should be fine.

However, there is another case where all the source urls are cloud urls and the destination is local path. In that case, it will tag the operation as CloudToCloud. If we pass the args as mentioned here https://github.com/GoogleCloudPlatform/gsutil/pull/986#discussion_r409175426 then perhaps we can check if the the last item `args[-1]` is a cloud path and at the same time there is a cloud item in the source list `args[:-1]`.
Another trivial optimization would be to only check the first and last element in args, as we are making the assumption that user is not going to mix local and cloud path in the src list.